gnssanalysis/gn_download.py

```diff
@@ -577,6 +822,58 @@ def download_file_from_cddis(
                 _time.sleep(_random.uniform(0.0, 2.0**retries))
 
 
+def download_file_from_cddis(
+    filename: str,
+    ftp_folder: str,
+    output_folder: _Path,
+    max_retries: int = 3,
+    decompress: bool = True,
+    if_file_present: str = "prompt_user",
+    note_filetype: str = None,
+) -> None:
+    """Downloads a single file from the cddis ftp server.
+
+    :param filename: Name of the file to download
+    :ftp_folder: Folder where the file is stored on the remote
+    :output_folder: Folder to store the output file
+    :ftps: Optional active connection object which is reused
+    :max_retries: Number of retries before raising error
+    :uncomp: If true, uncompress files on download
+    """
+    with ftp_tls("gdc.cddis.eosdis.nasa.gov") as ftps:
+        ftps.cwd(ftp_folder)
+        retries = 0
+        download_done = False
+        while not download_done and retries <= max_retries:
+            try:
+                download_filepath = attempt_ftps_download(
+                    download_dir=output_folder,
+                    ftps=ftps,
+                    filename=filename,
+                    type_of_file=note_filetype,
+                    if_file_present=if_file_present,
+                )
+                if decompress and download_filepath:
+                    download_filepath = decompress_file(
+                        input_filepath=download_filepath, delete_after_decompression=True
+                    )
+                download_done = True
+                if download_filepath:
+                    logging.info(f"Downloaded {download_filepath.name}")
+            except _ftplib.all_errors as e:
+                retries += 1
+                if retries > max_retries:
+                    logging.info(f"Failed to download {filename} and reached maximum retry count ({max_retries}).")
+                    if (output_folder / filename).is_file():
+                        (output_folder / filename).unlink()
+                    raise e
+
+                logging.debug(f"Received an error ({e}) while try to download {filename}, retrying({retries}).")
+                # Add some backoff time (exponential random as it appears to be contention based?)
+                _sleep(_random.uniform(0.0, 2.0**retries))
+    return download_filepath
```

Comment:
typehint of the return is `None` in the declaration

Reply:
You've actually highlighted that the function just before this has the exact same name: `download_file_from_cddis` :smile: 

I've deleted it now